docs/pages/deploy-a-cluster/teleport-enterprise/license.mdx

```diff
@@ -22,19 +45,106 @@ auth_service:
+
+Only Auth Service instances require the license. Instances of other Teleport
+services do not need a license file unless they are also running the Auth
+Service.
+
+### Check your license expiration date
+
+Your Teleport Enterprise license contains an X.509 certificate that contains the
+expiration date of your license. 
+
+Run the following command to inspect the expiration date of your license,
+assuming your license is saved as `license.pem`:
+
+```code
+$ openssl x509 -text -in license.pem | grep "Not After"
+             Not After : Dec 16 19:43:40 2022 GMT
 ```
 
-## License updates and renewals
+### Renew or update your license
+
+If you have changed your license agreement with Teleport, e.g., you have added
+or removed support for a feature, you must obtain your license file again and
+replace your existing license file on all Teleport Auth Service instances. You
+must also obtain a new license file after renewing your Teleport Enterprise
+license.
+
+Once you restart the Teleport Auth Service, any changes to the license will take
+effect.
+
+The Teleport Auth Service checks the expiration status of your license every
+hour. This means that, after you replace your license file and restart your Auth
+Service instances, it can take up to an hour to confirm the new license. 
+
+## When your license expires
+
+At 90 days before your Teleport Enterprise license expires, users will see an
+alert similar to the following when running `tsh login`, `tsh status`, or any
+`tctl command`:
+
+```text
+Your Teleport Enterprise Edition license will expire in 90 days on one or more
+of your auth servers. Please reach out to licenses@goteleport.com to obtain a
+new license. Inaction may lead to unplanned outage or degraded performance and
+support.
+```
+
+Users will see a similar message as a banner when accessing the Teleport Web UI:
+
+![License warning banner](../../../img/enterprise/license-warning.png)
+
+When your license expires, you will have a 30-day grace period before Teleport
+Enterprise features become disabled. 
+
+`tsh` users will see warnings indicating that Enterprise features are no longer
+available:
+
+```text
+Your Teleport Enterprise Edition license has expired on one or more of your auth
+servers. Please reach out to licenses@goteleport.com to obtain a new license.
+Inaction may lead to unplanned outage or degraded performance and support.
+```
+
+Teleport Web UI users will see an alert banner:
+
+![License expired banner](../../../img/enterprise/license-expired.png)
+
+Attempts to authenticate to Teleport via Single Sign-On connectors will fail.
+
+## Unlicensed Teleport features
+
+If users attempt to use a Teleport feature that your license does not allow,
+they will see an error message. For example, if you attempt to use Database
+Access without it being included in the license, you will get a message similar
+to:
+
+```text
+this Teleport cluster is not licensed for database access, please contact the cluster administrator
```

Comment:
We're starting to standardize the error messages for enterprise features. In most cases, it will look like this:

```text
SAML: this feature requires Teleport Enterprise
```